examples/generalized_linear_models/GLM-discrete-choice_models.myst.md

```diff
@@ -0,0 +1,772 @@
+
+$$ dU = \beta_{ic} dic + \beta_{oc} doc = 0 \Rightarrow 
+-\frac{dic}{doc}\mid_{dU=0}=\frac{\beta_{oc}}{\beta_{ic}}$$
+
+Our parameter estimates differ slightly from the reported estimates, but we agree the model is inadequate. We will show a number of Bayesian model checks to demonstrate this fact, but the main call out is that the parameter values for installation costs should probably be negative. It's counter-intuitive that a $\beta_{ic}$ increase in price would increase the utility of generated by the installation even marginally as here. Although we might imagine that some kind of quality assurance comes with price which drives satisfaction with higher installation costs. The coefficient for repeat operating costs is negative as expected. Putting this issue aside for now, we'll show below how we can incorporate prior knowledge to adjust for this kind of conflicts with theory. 
+
+But in any case, once we have a fitted model we can calculate the marginal rate of substitution as follows:
+
+```{code-cell} ipython3
+## marginal rate of substitution for a reduction in installation costs
+summaries["mean"]["beta_oc"] / summaries["mean"]["beta_ic"]
+```
+
+This statistic gives a view of the relative importance of the attributes which drive our utility measures. But being good Bayesians we actually want to calculate the posterior distribution for this statistic.
+
+```{code-cell} ipython3
+fig, ax = plt.subplots(figsize=(20, 10))
+
+ax.hist(
+    az.extract(idata_m1["posterior"]["beta_oc"] / idata_m1["posterior"]["beta_ic"])["x"],
+    bins=30,
+    ec="black",
+)
+ax.set_title("Uncertainty in Marginal Rate of Substitution \n Operating Costs / Installation Costs");
+```
+
+which suggests that there is almost twice the value accorded to the a unit reduction in recurring operating costs over the one-off installation costs. Whether this is remotely plausible is almost beside the point since the model does not even closely capture the data generating process. But it's worth repeating that the native scale of utility is not straightforwardly meaningful, but the ratio of the coefficients in the utility equations can be directly interpreted.  
+
+To assess overall model adequacy we can rely on the posterior predictive checks to see if the model can recover an approximation to the data generating process.
+
+```{code-cell} ipython3
+idata_m1["posterior"]["p"].mean(dim=["chain", "draw", "obs"])
+```
+
+```{code-cell} ipython3
+fig, axs = plt.subplots(1, 2, figsize=(20, 10))
+ax = axs[0]
+counts = wide_heating_df.groupby("depvar")["idcase"].count()
+predicted_shares = idata_m1["posterior"]["p"].mean(dim=["chain", "draw", "obs"])
+ci_lb = idata_m1["posterior"]["p"].quantile(0.025, dim=["chain", "draw", "obs"])
```

Comment:
With the mean it doesn't matter, but for the quantiles, you are now doing a different thing. I am not sure which one you want. Now you take quantiles over all 3 input dims, before you were taking the mean over samples first, then getting the _quantiles of these means_ over the obs dimension.

Reply:
Thanks @OriolAbril, yes i think it's the former quantity i want. I want to use it to create the interval on "p" and this way where i take the quantile over the same dimensions as i'm taking the mean puts them on the same scale, which i'm using in the graph to represent the interval.